### PR TITLE
Feature/already met backend

### PR DESCRIPTION
**Pull request by BotondAKovacs**
## Changes Made:

- Created a shared UI componenst ProfileListScreen.kt to modularise unblockScreen that is very similar to alreadyMetScreen
- Added a list to the Profile that tracks the already met users' profiles
- Added the methods to add remove already met user
- Check connectivity when user clicks on the already met page button in the settings and on the "i already met this person" button in the otherProfileView
- Filter out the already met users from the around you page

### CODE REVIEW

File: app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt [hunk 1]
```diff
@@ -378,6 +382,42 @@ open class ProfilesViewModel(
...
+   * @param uid The unique ID of the user being removed.
+   */
+  fun removeAlreadyMet(uid: String) {
+    _selfProfile.update { currentProfile ->
+      currentProfile?.copy(hasAlreadyMet = currentProfile.hasAlreadyMet.filter { it != uid })
+    }
+    updateProfile(selfProfile.value!!)
+    getBlockedUsers()
+  }
+
```
Comment (JanStaszewiczEPFL) [inline app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt:409]: You could make one update function for addAlreadyMet and removeAlreadyMet with something like : 

```
fun updateAlreadyMet(uid: String, add: Boolean) {
    _selfProfile.update { currentProfile ->
        currentProfile?.copy(
            hasAlreadyMet = if (add) {
                currentProfile.hasAlreadyMet + uid
            } else {
                currentProfile.hasAlreadyMet.filter { it != uid }
            }
        )
    }
} 
```

But I guess having both part separated can help us during testing and maintaining a clean code.

File: app/src/androidTest/java/com/github/se/icebreakrr/ui/sections/AlreadyMetTest.kt [hunk 2]
```diff
@@ -2,90 +2,178 @@ package com.github.se.icebreakrr.ui.sections
...
-  private fun mockProfile() =
-      Profile(
-          uid = "1",
-          name = "John Doe",
-          gender = Gender.MEN,
-          birthDate = birthDate2002,
-          catchPhrase = "Just a friendly guy",
-          description = "I love meeting new people.",
-          tags = listOf("friendly", "outgoing"))
 }
```
Comment (JanStaszewiczEPFL) [inline app/src/androidTest/java/com/github/se/icebreakrr/ui/sections/AlreadyMetTest.kt:179]: Great testing!

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -32,6 +32,12 @@
 
     <string name="Already_Met_Button_Text">"I Already Met This Person"</string>
     <string name="Already_Met_Settings_Button">People You Already Met</string>
+    <string name="Already_Met_Button_Success">You successfully added this user to your already met list</string>
+    <string name="no_already_met_users">You haven\'t met anyone yet. Go explore the world!</string>
+    <string name="unmeet_confirm">Did you add this user by mistake?</string>
+    <string name="unmeet_confirm_message">Remove from your already met list?</string>
+    <string name="unmeet_yes">Yes</string>
+    <string name="unmeet_no">No</string>
```
Comment (JanStaszewiczEPFL) [inline app/src/main/res/values/strings.xml:40]: Thanks for externalizing strings into strings.xml, it's clean code!

**Review by JanStaszewiczEPFL — APPROVED** [review]
Overall really good code, not a lot to say, I proposed a few changes but it is discussable so I approve.

File: app/src/androidTest/java/com/github/se/icebreakrr/ui/sections/AlreadyMetTest.kt (hunk 2 above)
Comment (BotondAKovacs) [reply app/src/androidTest/java/com/github/se/icebreakrr/ui/sections/AlreadyMetTest.kt:179]: Thanks!

**Pull request merged**